File: src/EventEngine.cpp
#include "EventEngine.h"
#include "Exceptions.h"
#include "Constants.h"

OP_NAMESPACE_START

void EventEngine::put(EVENT event) {
    _core.push(event);
};

EVENT EventEngine::get(void) {
    _check_core_empty();
    EVENT result = _core.front();

    _core.pop();
    return result;
};

int EventEngine::size(void) const {
    return _core.size();
};

void EventEngine::_check_core_empty(void) const {
    if (_core.empty()) {
        throw QueueEmptyError();
    };
};

bool EventEngine::is_core_empty(void) const {
    return _core.empty() ? true : false;
};
OP_NAMESPACE_END

File: lib/event_engine.py
import queue
from enum import Enum

from OnePy.constants import EVENT


class EventEngine(object):
    def __init__(self):
        self._core = queue.Queue()  # type:queue.Queue

    def put(self, event: EVENT):
        self._core.put(event)

    def get(self) -> EVENT:
        return self._core.get(block=False)

    def is_empty(self) -> bool:
        return self._core.empty()

class EVENT(Enum):
    Market_updated = 'Market_updated'
    Data_cleaned = 'Data_cleaned'
    Signal_generated = 'Signal_generated'
    Submit_order = 'Submit_order'
    Record_result = 'Record_result'
